### PR TITLE
[torch_glow] Add support for aten::copy_ op

### PR DESCRIPTION
**Pull request by sgeorgiev255**
Summary:

This PR adds support for loading the aten::copy/aten::copy_ ops in PyTorchModelLoader.

Test Plan:
Relevant tests have been added to copy_test.py.

### CODE REVIEW

File: torch_glow/src/PyTorchModelLoader.cpp [hunk 1]
```diff
@@ -4334,6 +4335,46 @@ Error PyTorchModelLoader::loadTranspose(const torch::jit::Node *ptNode) {
...
 }
 
+Error PyTorchModelLoader::loadCopy(const torch::jit::Node *ptNode) {
+  auto inputs = ptNode->inputs();
+  auto outputs = ptNode->outputs();
+
+  RETURN_IF_ERR(checkInputAndOutputSizes(inputs, 3, outputs, 1));
+
+  glow::NodeValue lhs, rhs;
+  ASSIGN_VALUE_OR_RETURN_ERR(lhs, getGlowNodeValueForValue(inputs[0]));
```
Comment (zrphercule) [inline torch_glow/src/PyTorchModelLoader.cpp:4345]: Would you mind giving every input an explicit name? Like CopyInput::lhs = 0, etc. You can reference from other ops.

File: torch_glow/src/PyTorchModelLoader.cpp [hunk 2]
```diff
@@ -4334,6 +4335,46 @@ Error PyTorchModelLoader::loadTranspose(const torch::jit::Node *ptNode) {
...
+
+  // Handle case where rhs needs to be broadcasted to match lhs
+  const bool needsBroadcast = lhs.getType()->dims() != rhs.getType()->dims();
+  std::vector<NodeValue> broadcastedInputs;
+  if (needsBroadcast) {
+    broadcastedInputs = F_.broadcastInputs(
+        /* axis */ -1, {lhs, needsConvertTo ? toNode->getResult() : rhs});
+  }
+
+  RETURN_ERR(addValueMapping(
```
Comment (zrphercule) [inline torch_glow/src/PyTorchModelLoader.cpp:4373]: Can we reorganize the logic here a little bit to evade using so much "...?...:..."? Like, having a temporary glow::NodeValue to store the intermediate result.

**Review by zrphercule — CHANGES_REQUESTED** [review]
Overall LGTM, please address the comments and we will be able to land it.
Thanks!

File: torch_glow/src/PyTorchModelLoader.cpp (hunk 1 above)
Comment (sgeorgiev255) [reply torch_glow/src/PyTorchModelLoader.cpp]: Done.

File: torch_glow/src/PyTorchModelLoader.cpp (hunk 2 above)
Comment (sgeorgiev255) [reply torch_glow/src/PyTorchModelLoader.cpp]: Done.

**Comment by zrphercule** [comment]
@sgeorgiev255  It seems the unit test of copy_op is broken in CI, would you mind taking a look?

**Comment by sgeorgiev255** [comment]
> @sgeorgiev255 It seems the unit test of copy_op is broken in CI, would you mind taking a look?

Hi, I wasn't able to reproduce the failing test locally until I updated to latest master of pytorch. I can now reproduce it but I am not sure what is causing this issue, seems like a regression in pytorch?

**Comment by sgeorgiev255** [comment]
@zrphercule I have rebased this PR and took another look at the failing test. Running a simple test in the python interpreter shows this result:

```
>>> a = torch.zeros(2, 3, 4, dtype=torch.float32)
>>> b = torch.randn(4, dtype=torch.float16)
>>> a.copy_(b)
tensor([[[-8.5303e-01, -3.4961e-01,  1.2764e+00, -2.7168e+00],
         [ 3.4102e+00, -3.9340e+03,  3.1643e-03, -6.9760e+03],
         [ 1.4400e+04, -2.3483e-02,         nan,  0.0000e+00]],

        [[ 3.8743e-06,  0.0000e+00,  0.0000e+00,  0.0000e+00],
         [-8.7891e-03, -3.8281e+01,         nan,  0.0000e+00],
         [ 1.5216e+04,  1.3176e+04,  7.2960e+03,  2.2800e+03]]])
```

So I think it is an issue with pytorch? I am running latest master from today (commit hash `ad7d208ba5f2c5614679a7999918b75ae74530e9`).

**Comment by zrphercule** [comment]
> @zrphercule I have rebased this PR and took another look at the failing test. Running a simple test in the python interpreter shows this result:
> 
> ```
> >>> a = torch.zeros(2, 3, 4, dtype=torch.float32)
> >>> b = torch.randn(4, dtype=torch.float16)
> >>> a.copy_(b)
> tensor([[[-8.5303e-01, -3.4961e-01,  1.2764e+00, -2.7168e+00],
>          [ 3.4102e+00, -3.9340e+03,  3.1643e-03, -6.9760e+03],
>          [ 1.4400e+04, -2.3483e-02,         nan,  0.0000e+00]],
> 
>         [[ 3.8743e-06,  0.0000e+00,  0.0000e+00,  0.0000e+00],
>          [-8.7891e-03, -3.8281e+01,         nan,  0.0000e+00],
>          [ 1.5216e+04,  1.3176e+04,  7.2960e+03,  2.2800e+03]]])
> ```
> 
> So I think it is an issue with pytorch? I am running latest master from today (commit hash `ad7d208ba5f2c5614679a7999918b75ae74530e9`).

This is weird... Could you please add an issue here for us to investigate? https://github.com/pytorch/pytorch/issues
Thanks!

**Comment by sgeorgiev255** [comment]
Commenting so the PR doesn't get closed.

**Pull request closed without merging**